src/UI/Actions.hs

```diff
@@ -832,17 +832,22 @@ chain (Action d1 f1) (Action d2 f2) = Action (d1 <> d2) (f1 >=> f2)
 --
 chain'
     :: forall ctx ctx' a v v'.
-       (HasName ctx', HasViewName v', ViewTransition v v')
+       (HasName ctx, HasViewName v, HasName ctx', HasViewName v', ViewTransition v v')
     => Action v ctx AppState
     -> Action v' ctx' a
     -> Action v ctx a
 chain' (Action d1 f1) (Action d2 f2) =
   Action (d1 <> d2) (f1 >=> switchMode >=> f2)
   where
-    switchMode s = pure $ s &
-      transitionHook (Proxy :: Proxy v) (Proxy :: Proxy v')
-      . over (asViews . vsFocusedView) (Brick.focusSetCurrent (viewname (Proxy :: Proxy v')))
-      . set (asViews . vsViews . at (viewname (Proxy :: Proxy v')) . _Just . vFocus) (name (Proxy :: Proxy ctx'))
+    switchMode s = do
+      liftIO . view (asConfig . confLogSink) s . T.pack $
+        "chain' "
+          <> show (viewname (Proxy :: Proxy v)) <> "/" <> show (name (Proxy :: Proxy ctx)) <> " -> "
+          <> show (viewname (Proxy :: Proxy v')) <> "/" <> show (name (Proxy :: Proxy ctx'))
+      pure $ s &
+        transitionHook (Proxy :: Proxy v) (Proxy :: Proxy v')
+        . over (asViews . vsFocusedView) (Brick.focusSetCurrent (viewname (Proxy :: Proxy v')))
```

Comment:
I welcome this, because this could make quite a bit of code redundant I've been dragging around. However there is an additional bit missing, which set's the widgets state in the view to visible. Since the widgets sit in a list of lists, the code would need an O(n) operation to find the widget's "Layer" or ... well use a better data structure. Have a look at some of the `Focusable` instances. Not sure whether you're keen ...

Reply:
What code will it make redundant?

Any enhancements/additions can wait for another PR.  I don't did the bare minimum I needed, but you are the UI author and have a better idea what would be worth logging.

Reply:
Fair enough!